WordPress/src/main/java/org/wordpress/android/models/recommend/RecommendApiCallsProvider.kt

```diff
@@ -8,89 +8,117 @@ import com.wordpress.rest.RestRequest.ErrorListener
+        val notNullMessage = logMessage ?: "logErrorAndTrack > logMessage was null"
+        AppLog.d(T.API, notNullMessage)
+        analyticsUtilsWrapper.trackRecommendAppFetchFailed(source, notNullMessage)
     }
 
-    private fun getTemplateFromJson(json: JSONObject?, appName: String): RecommendCallResult {
-        return json?.let {
-            val name = it.optString("name")
+    private fun getTemplateFromJson(
+        json: JSONObject?,
+        appName: String,
+        source: RecommendAppSource
+    ): RecommendCallResult {
+        return if (json != null) {
+            val name = json.optString("name")
             if (name == appName) {
                 try {
                     val gson = Gson()
                     val mapType = object : TypeToken<RecommendTemplateData>() {}.type
-                    val template = gson.fromJson<RecommendTemplateData>(it.toString(), mapType)
+                    val template = gson.fromJson<RecommendTemplateData>(json.toString(), mapType)
                     AppLog.d(
                             T.API,
                             "getTemplateFromJson > name[${template.name}], " +
                                     "link[${template.link}], message[${template.message}]"
                     )
                     Success(template)
                 } catch (jsonEx: JsonParseException) {
-                    AppLog.d(
-                            T.API,
-                            "getTemplateFromJson > Error parsing server API response: error[{${jsonEx.message}}]"
-                    )
+                    val logMessage = "getTemplateFromJson > Error parsing server API" +
+                            " response: error[{${jsonEx.message}} json[$json]]"
+                    logErrorAndTrack(source, logMessage)
                     Failure(
                             contextProvider.getContext()
                                     .getString((R.string.recommend_app_bad_format_response))
                     )
                 }
             } else {
-                AppLog.d(T.API, "getTemplateFromJson > wrong app name received: expected[$appName] got[$name]")
+                val logMessage = "getTemplateFromJson > wrong app name received: expected[$appName] got[$name]"
+                logErrorAndTrack(source, logMessage)
                 Failure(contextProvider.getContext().getString(R.string.recommend_app_bad_format_response))
             }
-        } ?: Failure(contextProvider.getContext().getString(R.string.recommend_app_null_response))
+        } else {
+            val logMessage = "getTemplateFromJson > null response received"
+            logErrorAndTrack(source, logMessage)
+            Failure(contextProvider.getContext().getString(R.string.recommend_app_null_response))
+        }
     }
 
-    private fun getErrorStringAndLog(
-        functionName: String,
+    private fun getNetErroAndLogStrings(
```

Comment:
Small typo `Erro` → `Error` 

(that would be the correct spelling in Portuguese, though 😆)